### PR TITLE
[BUGFIX stable] Allow `store` to be injected on `Route`

### PR DESCRIPTION
**Pull request by chriskrycho**
In the public types, TypeScript (correctly) complains when users try to inject the store with `@service store` on a `Route` subclass, because the item is an accessor on `Route` itself, and the service injection "happens" to work only because the `set store(value)` smashes itself as well as the corresponding `get store()` using `defineProperty`.

Work around this by renaming the getter in question to `_store` and updating the one usage to check for a `store` property before falling back to the `_store` property. Happily, this also dovetails with the plan to deprecate the automatic `findModel` behavior this exists to support: once that is removed, this entire code path can be cleaned up trivially.

### CODE REVIEW

File: tests/docs/expected.js [hunk 1]
```diff
@@ -484,7 +484,7 @@ module.exports = {
     'sort',
     'sortBy',
     'startRouting',
-    'store',
+    '_store',
```
Comment (gitKrystan) [inline tests/docs/expected.js:487]: ELI5 this file?

Comment (chriskrycho) [reply tests/docs/expected.js:487]: Every name Ember exports gets checked here as a suuuuper basic "smoke test" to make sure we don't accidentally stop publishing something.

Comment (chriskrycho) [reply tests/docs/expected.js]: Also, there is apparently some *other* usage of `store` so I have reverted this change. 😂

**Review by wagenet — APPROVED** [review]
